sharpy/aero/models/aerogrid.py

```diff
@@ -270,15 +265,28 @@ def generate_zeta_timestep_info(self, structure_tstep, aero_tstep, beam, setting
                             raise NotImplementedError(str(self.data_dict['control_surface_type'][i_control_surface]) +
                                 ' control surfaces are not yet implemented')
 
-
+                # add sweep for aerogrid warping in constraint defintition
+                # if no constraint_xx.aerogrid warp factor is provided, this will be ignored
+                ang_warp = 0.
+                if structure_tstep.mb_dict is not None:
+                    if structure_tstep.mb_prescribed_dict is not None:
+                        for i_constraint in range(structure_tstep.mb_dict['num_constraints']):
+                            try:
+                                cst_name = f"constraint_{i_constraint:02d}"
+                                ctrl_id = structure_tstep.mb_dict[cst_name]['controller_id'].decode('UTF-8')
+                                f_warp = structure_tstep.mb_dict[cst_name]['aerogrid_warp_factor'][i_elem, i_local_node]
+                                ang_z = structure_tstep.mb_prescribed_dict[ctrl_id]['delta_psi'][2]
+                                ang_warp += f_warp * ang_z
+                            except KeyError:
+                                continue
```

Comment:
The constraints here are input as dictionaries and have different key-value pairs depending on their functionality (hinge axis, node number etc.). Warping the aerodynamic grid will occur here if a constraint has both ```controller_id``` and ```aerogrid_warp_factor``` entries, with the intended behavior to skip this code if a constraint is missing one. A key error will occur if it's a constraint that is missing either of these entries (and therefore is not a constraint which requires the warping). Of course, it could technically fail if I had coded it wrong and ```delta_psi``` is not defined, and this would incorrectly ignore this code, but I'm pretty sure that can't happen.